### PR TITLE
Changes to the README file

### PR DESCRIPTION
**Pull request by Hammarn**
General updates to the README file. Now all programs in the pipeline should have some description of their use and what kinda of a result to expect. I also tried to include what a good and bad result would be (something that is surprisingly lacking in many manuals). 
Changed the images to be from the MultiQC report instead. 

Question? Should we add a list of all the files generated by the pipeline to this file and what they do/are useful for?

### CODE REVIEW

File: Docs/README.md [hunk 1]
```diff
@@ -1,10 +1,13 @@
 #Manual RNA BP 2.0
-This is an an attempt at descripting the different programs and analyses that consitutes the NGI RNA BP 2.0 analysis pipeline. 
+This is an an attempt at descripting the different programs and analyses that consitutes the NGI RNA BP 2.0 analysis pipeline and to inform the use of what to look for when analysing their samples. 
+
+The programs used in this pipeline all have their own outputs, here we will mostly describe what the output looks like in the MultiQC report that is genrated at the end of the pipeline.
```
Comment (ewels) [inline Docs/README.md]: typo: _generated_

File: Docs/README.md [hunk 2]
```diff
@@ -15,11 +18,12 @@ This is an an attempt at descripting the different programs and analyses that co
...
-* preseq - subread featureCounts - gene counts, biotype counts, rRNA estimation.
+* dupRadar - investigate duplication problems
+* preseq - 
+* subread featureCounts - gene counts, biotype counts, rRNA estimation.
 * String Tie - FPKMs for genes and transcripts
 * edgeR - create MDS plot and sample pairwise distance heatmap / dendrogram
-* MultiQC
+* MultiQC - aggregate report
 
 ## FastQC
```
Comment (ewels) [inline Docs/README.md]: Documentation link for FastQC is unofficial - better to link to the official (and up to date) help pages: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/

File: Docs/README.md [hunk 3]
```diff
@@ -35,6 +39,10 @@ The NGI RNA BP 2.0 pipeline uses TrimGalore for removal of adapter contamination
 ## STAR
 STAR is a read aligner designed for RNA sequencing.  STAR stands for Spliced Transcripts Alignment to a Reference, and it is designed with speed in mind. 
 
+The STAR section of the MultiQC report is rather self explanatory the more blue (Uniquely mapped or Mapped to multiple loci) the better conversely, the more pink or burgundy (Unmapped), the worse the alignment.
```
Comment (ewels) [inline Docs/README.md]: _better. Conversely,_

File: Docs/README.md [hunk 4]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
 Reads mapped in proper pairs:                  21856264
 Proper-paired reads map to different chrom:    7648
 ```
+The MultiQC plot of this information can be a bit overwhelming at first but it displayes the above information simultaneously for all the samples in your report. With each sample represented as a dot.
```
Comment (ewels) [inline Docs/README.md]: _MultiQC plots each of these statistics in a dot plot (each sample is a dot - hover to see the sample highlighted across all fields)._

File: Docs/README.md [hunk 5]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
 Reads mapped in proper pairs:                  21856264
 Proper-paired reads map to different chrom:    7648
 ```
+The MultiQC plot of this information can be a bit overwhelming at first but it displayes the above information simultaneously for all the samples in your report. With each sample represented as a dot. 
 ###infer_experiment.py
 
-This scripts guesses how the reads were stranded for strand specific RNA-seq data. Example output:
+This scripts guesses how the reads were stranded for strand specific RNA-seq data.
```
Comment (ewels) [inline Docs/README.md]: _This script_

File: Docs/README.md [hunk 6]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+Fraction of reads explained by "1+-,1-+,2++,2--": 0.4752
 ```
 
+
+
 ###junction_saturation.py
 This scripts aims to determine if the used sequencing depth is enough to perform  alternative splicing analysis on the data. 
 
-![Junction Saturation](images/junction_saturation.png)
+![Junction Saturation](images/Junction_saturation.png)
```
Comment (ewels) [inline Docs/README.md]: Revert this change? Image isn't found.

File: Docs/README.md [hunk 7]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
-Fraction of reads explained by "1+-,1-+,2++,2--": 0.4925
+Fraction of reads failed to determine: 0.0409
+Fraction of reads explained by "1++,1--,2+-,2-+": 0.4839
+Fraction of reads explained by "1+-,1-+,2++,2--": 0.4752
 ```
 
+
+
 ###junction_saturation.py
 This scripts aims to determine if the used sequencing depth is enough to perform  alternative splicing analysis on the data.
```
Comment (ewels) [inline Docs/README.md]: _This script_

File: Docs/README.md [hunk 8]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 This scripts aims to determine if the used sequencing depth is enough to perform  alternative splicing analysis on the data. 
 
-![Junction Saturation](images/junction_saturation.png)
+![Junction Saturation](images/Junction_saturation.png)
 
-In the above example output the red line (known junction) has reached a plateau. Nearly all known junctions in this data set have already been detected, and deeper sequencing will thus not help in detecting more known junctions. The green line (novel junction) has not plateaued and thus further sequencing will likely aid in junction saturation.
-In summary, if the lines are flattening out then you have sequenced deeply enough. 
-###RPKM saturation
+
+A line that has plateaued indicates that all junctions have been detected and that further secuencing will not yield more results. Insteads the same fragment would be sequenced again and again. None of the lines in the example above have plataued and thus these samples could reveal more alternative splicing information if they were sequenced deeper.
```
Comment (ewels) [inline Docs/README.md]: _indicates that all junctions have been detected_ - this is not true. It means that the sample has been sequenced to saturation. It could be a very tiny / low complexity sample with hardly any junctions represented in the library (different to detecting _all_ junctions).

> The x-axis shows each library at different down-sampling percentages. If the library has been sequenced to saturation, the line will plateau and increasing sequencing depth will not yield any additional splicing information.

File: Docs/README.md [hunk 9]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
-In summary, if the lines are flattening out then you have sequenced deeply enough. 
-###RPKM saturation
+
+A line that has plateaued indicates that all junctions have been detected and that further secuencing will not yield more results. Insteads the same fragment would be sequenced again and again. None of the lines in the example above have plataued and thus these samples could reveal more alternative splicing information if they were sequenced deeper. 
+
+###RPKM_saturation.py
 Resamples a subset of the total RNA reads and calculates the RPKM value for each subset. We use the default subsets of every 5% of the total reads. I.e 5%,10%...95%,100%. 
 A percent relative error is then calcuated based on the subsamples, this is the y-axis in the graph. 
+As of this text being written there isn't a MultiQC submodule for displaying the output of RPKM_saturation.py. Below is instead the pdfs generated by the script itself:
```
Comment (ewels) [inline Docs/README.md]: This isn't super clear.. Maybe instead:

> The variance and error of RNA-seq transcript quantification varies according to level of transcription. This script divides all transcripts into quarters based on RPKM percentile and then measures the _percent relative error_ at increasing levels of downsampling. If the sample is sequenced to a sufficient depth, the RPKM will be reproducible at varying subsample levels and the percentage relative error will be low.

File: Docs/README.md [hunk 10]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 
 <img src= "images/saturation.png" width="500" height="500" >
 
 
-###read duplication
+###read_duplication.py
+This script investigates the read duplication rate. A good library will generally have a slightly declining graph. The red library in the example graph really stands out and needs investigating.
```
Comment (ewels) [inline Docs/README.md]: > This plot shows the number of reads (y-axis) with a given number of exact duplicates (x-axis). Most reads in an RNA-seq library should have a low number of exact duplicates. Samples which have many reads with many duplicates (a large area under the curve) may be suffering excessive technical duplication.
>
> The red line here shows a sample with a very low aligned read count, so it has very reads and few duplicates.

File: Docs/README.md [hunk 11]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 
-###read duplication
+###read_duplication.py
+This script investigates the read duplication rate. A good library will generally have a slightly declining graph. The red library in the example graph really stands out and needs investigating. 
+
+<img src= "images/rseqc_read_dups_plot.png" >
+
+###inner_distance.py
+inner_distance.py  tries to calculate the inner distance (or insert size) between two paired RNA reads. Calculated as _start of reads 2_ - _end of read 2_. 
+Note that values can be negative if the reads overlap, this is indicative of a poor library, and fragmented RNA. As in the example picture below.
```
Comment (ewels) [inline Docs/README.md]: _this is indicative of a poor library_ - not necessarily, it can be caused by a lot of different things.

> This script calculates the inner distance (or insert size) between two paired RNA reads. Note that this distance does _not_ include the sequenced read lengths, so numbers can be negative if the reads overlap.
>
> The library shown below has very short fragments, with most paired reads overlapping.

File: Docs/README.md [hunk 12]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+<img src= "images/rseqc_read_dups_plot.png" >
+
+###inner_distance.py
+inner_distance.py  tries to calculate the inner distance (or insert size) between two paired RNA reads. Calculated as _start of reads 2_ - _end of read 2_. 
+Note that values can be negative if the reads overlap, this is indicative of a poor library, and fragmented RNA. As in the example picture below. 
 
+<img src= "images/rseqc_inner_distance_plot.png">
 
-<img src= "images/read_duplication.png" width="500" height="500" >
+###gene\_body_coverage.py
```
Comment (ewels) [inline Docs/README.md]: Missing any content here?

File: Docs/README.md [hunk 13]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 
-###inner distance
 
-###gene body coverage
+###read_distribution.py
+read_distribution.py calculates how mapped reads are distributed over genome features. A good result for a standard RNA seq experiments is genreally to have as much exonic reads as possible, i.e CDS_Exons near 100%. A large amount of intronic reads could be indicitative of DNA contamination in your sample or some other problem.
```
Comment (ewels) [inline Docs/README.md]: _as many_ exonic reads as possible - `CDS_Exons` should be close to 100%.

File: Docs/README.md [hunk 14]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 
-###read distribution
+<img src="images/rseqc_read_distribution_plot.png">
 
-###junction annotation
 
+###junction_annotation.py
+MultiQC module under way. Rscript included. Source it to generate a pie chart over the known and novel splicing junctions.
```
Comment (ewels) [inline Docs/README.md]: Do we not deliver the PDF already? We probably should do if we don't.

File: Docs/README.md [hunk 15]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
-###read distribution
+<img src="images/rseqc_read_distribution_plot.png">
 
-###junction annotation
 
+###junction_annotation.py
+MultiQC module under way. Rscript included. Source it to generate a pie chart over the known and novel splicing junctions. 
 ##dupRadar
 dupRadar is A Bioconductor library for R. 
 It plots the duplication rate per gene. A good sample (i.e. without PCR duplication?) will have a sigmodial shape as seen in the example output below:
```
Comment (ewels) [inline Docs/README.md]: No example.

File: Docs/README.md [hunk 16]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+###junction_annotation.py
+MultiQC module under way. Rscript included. Source it to generate a pie chart over the known and novel splicing junctions. 
 ##dupRadar
 dupRadar is A Bioconductor library for R. 
 It plots the duplication rate per gene. A good sample (i.e. without PCR duplication?) will have a sigmodial shape as seen in the example output below:  
 
-##preseq 
+##preseq
+Preseq estimates the complexity of a library, showing how many additional unique reads are sequenced for increasing the total read count. A shallow curve indicates that the libary has reached complexity saturation and further sequencing would likely not add further unique reads. The dashed line shows a perfectly complex library where total reads = unique reads. 
+<img src="images/preseq_complexity_curve.png">
```
Comment (ewels) [inline Docs/README.md]: Nice 👍  Again, example graph is kind of busy.. ;)

File: Docs/README.md [hunk 17]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 
-subread featureCounts - gene counts, biotype counts, rRNA estimation.
+##subread featureCounts 
+The featureCounts program from the subread package summarises the read distribution over genomic features such as genes, exons, promotors, gene bodies, genomic bins and chromosomal locations. 
+
+In addition we also plots the Biotype information, which includes miRNA, rRNA, diffferent kinds of pseudogenes to mention a few.
```
Comment (ewels) [inline Docs/README.md]: _we also plot_

File: Docs/README.md [hunk 18]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+##preseq
+Preseq estimates the complexity of a library, showing how many additional unique reads are sequenced for increasing the total read count. A shallow curve indicates that the libary has reached complexity saturation and further sequencing would likely not add further unique reads. The dashed line shows a perfectly complex library where total reads = unique reads. 
+<img src="images/preseq_complexity_curve.png">
 
-subread featureCounts - gene counts, biotype counts, rRNA estimation.
+##subread featureCounts 
+The featureCounts program from the subread package summarises the read distribution over genomic features such as genes, exons, promotors, gene bodies, genomic bins and chromosomal locations. 
+
+In addition we also plots the Biotype information, which includes miRNA, rRNA, diffferent kinds of pseudogenes to mention a few. 
+In both plot you generally want the blue bar to be as big ass possible. That means assigned feature in the general featureCounts plot and protein coding in the Biotype plot. The biotype plot is a useful tool for detecting rRNA contamination in your samples.
```
Comment (ewels) [inline Docs/README.md]: > Good RNA samples typically have most reads categorised as _Assigned_ - that is, overlapping a known genomic feature. Overlapping biotypes should mostly be _protein coding_ genes. The biotype plot can vary according to library preparation method and can highlight rRNA contamination.

File: Docs/README.md [hunk 19]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+
+In addition we also plots the Biotype information, which includes miRNA, rRNA, diffferent kinds of pseudogenes to mention a few. 
+In both plot you generally want the blue bar to be as big ass possible. That means assigned feature in the general featureCounts plot and protein coding in the Biotype plot. The biotype plot is a useful tool for detecting rRNA contamination in your samples. 
+
+<img src="images/featureCounts_assignment_plot.png">
+<img src="images/featureCounts_biotype_plot.png">
+ 
 ##String Tie
-FPKMs for genes and transcripts
+StringTie is a fast and highly efficient assembler of RNA-Seq alignments into potential transcripts. It uses a novel network flow algorithm as well as an optional de novo assembly step to assemble and quantitate full-length transcripts representing multiple splice variants for each gene locus.
```
Comment (ewels) [inline Docs/README.md]: This sounds very copy+pasted ;) Also the markdown formatting is a bit dodgy.

> StringTie assembles RNA-Seq alignments into potential transcripts. It assembles and quantitates full-length transcripts representing multiple splice variants for each gene locus.
>
> StringTie outputs FPKM metrics for genes and transcripts. The analysis generates the following files: 
> * `<sample>_Aligned.sortedByCoord.out.bam_transcripts.gtf`
>  * ### EXPLAIN WHAT THIS IS HERE ###
> * `<sample>_Aligned.sortedByCoord.out.bam.cov_refs.gtf`
>  * ### EXPLAIN WHAT THIS IS HERE ###

File: Docs/README.md [hunk 20]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+The output from stringtie is FPKMs for genes and transcripts. The analysis generates the following files: 
+_\<sample\>_ligned.sortedByCoord.out.bam_transcripts.gtf
+_\<sample\>_Aligned.sortedByCoord.out.bam_transcripts.gtf
+_\<sample\>_Aligned.sortedByCoord.out.bam.cov_refs.gtf
+
 ## edgeR
-edgR is a bioconductor package for R and is used to do differntial extression analysis on RNAseq data.
+edgR is a bioconductor package for R and is used to do differential extression analysis on RNAseq data.
```
Comment (ewels) [inline Docs/README.md]: This paragraph is a bit mental.

> edgeR is a bioconductor package for R used for RNAseq data analysis. The script included in the pipeline uses edgeR to normalise read counts and create a heatmap / dendrogram showing pairwise euclidean distance (sample similarity). It also creates a 2D MDS scatter plot showing sample grouping.

File: Docs/README.md [hunk 21]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
 Rcreate MDS plot and sample pairwise distance heatmap / dendrogram
 [edgeR Bioconductor page](https://bioconductor.org/packages/release/bioc/html/edgeR.html)
 The output generated from edgeR process is a heatmap and a MDS-plot.
-The heatmap may look something like this:
-<img src= "images/heatmap.png" width="500" height="500" >
+The heatmap may look something like this;
```
Comment (ewels) [inline Docs/README.md]: Can drop this line.

File: Docs/README.md [hunk 22]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+
+**Heatmap:**
+<img src= "images/mqc_hcplot_hocmzpdjsq.png" >
+**MDS plot:**
+<img src= "images/mqc_hcplot_ltqchiyxfz.png" >
+The hatmap and MDS plot together try and give you a feeling for which of your sample libraries that are most simmilar to eah other.
```
Comment (ewels) [inline Docs/README.md]: Can drop this line too I think, it's pretty obvious.

File: Docs/README.md [hunk 23]
```diff
@@ -64,65 +72,101 @@ Splice reads:                                  4161455
...
+
+**Heatmap:**
+<img src= "images/mqc_hcplot_hocmzpdjsq.png" >
+**MDS plot:**
+<img src= "images/mqc_hcplot_ltqchiyxfz.png" >
+The hatmap and MDS plot together try and give you a feeling for which of your sample libraries that are most simmilar to eah other. 
 
 ##MultiQC
-[MultiQC](http://multiqc.info/) is a visualation tool and is what we at NGI use to generate our hmtl reports. It collects all the reports for your samples and genearate clear and simple reports for all your samples for an easy overview of your project. All the results of the RNA seq BP pipeline 2.0 is available in the `multiqc_report.html` file.  
+[MultiQC](http://multiqc.info/) is a visualation tool and is what we at NGI use to generate our hmtl reports. It collects all the reports for your samples and genearate clear and simple reports for all your samples for an easy overview of your project. Almost all of the output results of the RNA seq BP pipeline 2.0 is available in the `multiqc_report.html` file.
```
Comment (ewels) [inline Docs/README.md]: A few typos, reads a bit strangely.. Also the filename will probably be different because it'll have the project name.

> [MultiQC](http://multiqc.info) is a visualisation tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available in within the report data directory.

File: Docs/README.md (hunk 11 above)
Comment (Hammarn) [inline Docs/README.md]: how about:

>One of the _potential_ causes is poor library, and fragmented RNA.

File: Docs/README.md (hunk 12 above)
Comment (Hammarn) [inline Docs/README.md]: yeah, I seem to have missed this one

File: Docs/README.md (hunk 14 above)
Comment (Hammarn) [inline Docs/README.md]: Hmm, it looks like we actually do. My bad

File: Docs/README.md (hunk 11 above)
Comment (ewels) [inline Docs/README.md]: Yeah, fine 👍

**Comment by ewels** [comment]
Bra jobbat!
